broker/lib/fabrik/LockStatusPoller.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const {
+  Etcd3,
+  EtcdLockFailedError
+} = require('etcd3');
+
+const config = require('../config');
+const logger = require('../logger');
+
+const client = new Etcd3({
+  hosts: config.etcd.url,
+  credentials: {
+    rootCertificate: Buffer.from(config.etcd.ssl.ca, 'utf8'),
+    privateKey: Buffer.from(config.etcd.ssl.key, 'utf8'),
+    certChain: Buffer.from(config.etcd.ssl.crt, 'utf8')
+  }
+});
+const time_minute = (1 * 60 * 1000);
+
+class LockStatusPoller {
+  constructor(opts) {
+    this.timeInterval = opts.time_interval || time_minute;
+    this.lock = null;
+  }
+
+  /**
+   * Name of the lock resource for the timer
+   */
+  get lockName() {
+    return `pollers/${this.constructor.name}`;
+  }
+
+  /**
+   * Gets the interval object created by the NodeJS engine for the interval timer
+   */
+  get timer() {
+    return this.interval;
+  }
+
+  /**
+   * Fetches the current etcd lock
+   */
+  get currentLock() {
+    return this.lock;
+  }
+
+  /**
+   * Abstract function to be implemented by subclasses
+   * This function must return a Bluebird promise
+   */
+  action() {
+    throw new Error('Not implemented in subclass');
+  }
+
+  /**
+   * Starts the timer
+   */
+  start() {
+    if (this.interval) {
+      throw new Error('timer already started');
+    }
+    this.interval = setInterval(() => {
+      this.lock = client.lock(this.lockName);
+      this.lock.do(() => {
```

Comment:
everytime you are getting the lock, wont it reset the reference in LN 64? so how does this truly work in HA setup? May be  am not aware of how etcd lock works. If you can add some comments to this code here, it will help in readability on what the above is trying to achieve.

Reply:
Actually I want to reset the reference so that I forcibly can release the lock in the `stop` function below